### ISSUE TITLE
Crash on attempt to link desktop app

### ISSUE THREAD

**Issue by cajopa** [signalapp/Signal-Android#9019]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Android app crashes when attempting to link desktop app.

### Steps to reproduce
- Open the desktop app. It is not currently linked, so it displays a QR code for linking.
- In the Android app, version 4.46.2, navigate to Settings -> Link and click the + button.
- Aim the camera at the QR code on screen.

**Actual result:**
As soon as the QR code is in frame, the Android app crashes, saying "Unfortunately, Signal has stopped.".
**Expected result:**
Add the desktop app identified by the QR code to linked devices.

### Device info

**Device:** Motorola Moto X Pure Edition (XT1575)
**Android version:** 6.0
**Signal version:** 4.46.2

### Link to debug log
https://debuglogs.org/4522f54ca6b4f79f5bf5d2d66ac355e716eefc240d338f51d74ac128865b95e8

**Comment by greyson-signal** [comment]
Huh, that's weird. Seems to be an exception in the QR-parsing library we use.

```java
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: java.lang.ArrayIndexOutOfBoundsException: length=146; index=146
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at com.google.zxing.qrcode.decoder.BitMatrixParser.readCodewords(BitMatrixParser.java:192)
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at com.google.zxing.qrcode.decoder.Decoder.decode(Decoder.java:149)
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at com.google.zxing.qrcode.decoder.Decoder.decode(Decoder.java:93)
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:78)
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at org.thoughtcrime.securesms.qr.ScanningThread.getScannedData(ScanningThread.java:113)
2019-09-05 10:18:59.276 CDT E UncaughtExceptionLogger: 	at org.thoughtcrime.securesms.qr.ScanningThread.run(ScanningThread.java:74)
```

**Comment by cajopa** [comment]
Is the library packaged with Signal or is it a system dependency? Forgive me if I am unfamiliar with how Android package dependencies work.

**Comment by cajopa** [comment]
Confirmed fixed in Signal 4.47.4 - thank you!

**Comment by campbe13** [comment]
I know this is closed but I am getting the same problem, I open the QR  URL and it crashes.  
Signal 4.60.0 any suggestions ?
